### PR TITLE
Adding NetCoreApp - Part 02

### PR DESCRIPTION
**Pull request by eddynaka**
Refs #1769

- Adding netcoreapp1.1, 2.1 and 3.1 to **EventSourceListener.Tests**
- Update **EtwCollector.Tests** to be multitarget
- Update **HostingStartup.Tests** to be multitarget
- Adding netcoreapp2.1 and 3.1 to **ILogger.Tests**
- Adding netcoreapp2.1 and 3.1 to **EventCounterCollector.Tests**

### CODE REVIEW

File: LOGGING/test/EtwCollector.Tests/EtwTelemetryModuleTests.cs [hunk 1]
```diff
@@ -487,7 +492,11 @@ public async Task ReportsHierarchicalActivities()
                     && string.Equals(eventName, "Request/Stop", StringComparison.Ordinal));
                 Assert.IsNotNull(requestStopEvent, "Request/Stop event not found");
                 Assert.IsTrue(requestStopEvent.Properties.TryGetValue("ActivityID", out string activityID), "Event does not have ActivityID property");
+#if NETCOREAPP
```
Comment (eddynaka) [inline LOGGING/test/EtwCollector.Tests/EtwTelemetryModuleTests.cs:495]: @TimothyMothra , don't know if that's expected. But, for netcoreapp, the guid has a very different value, generating an activityID != 0. So, changing this, the test started to pass.

**Comment by eddynaka** [comment]
@TimothyMothra , and, I would like to know if we can update the nuget MSTest.TestFramework to the newer version. In my VS, when i try to run net45 and net46 they dont work. But, after updating just this package they run normally

File: WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,41 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <RootNamespace>Microsoft.ApplicationInsights.HostingStartup.Tests</RootNamespace>
+    <AssemblyName>Microsoft.ApplicationInsights.HostingStartup.Tests</AssemblyName>
+    <TargetFrameworks>net45;netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
```
Comment (cijothomas) [inline WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj:6]: For hostingstartup, target only net45. This project is only consumed in .NET Framework environment.

File: LOGGING/test/EtwCollector.Tests/EtwCollector.Tests.csproj [hunk 3]
```diff
@@ -2,9 +2,10 @@
 
   <PropertyGroup>
     <RootNamespace>Microsoft.ApplicationInsights.EtwCollector.Tests</RootNamespace>
-    <AssemblyName>Microsoft.ApplicationInsights.EtwCollector.Net46.Tests</AssemblyName>
-    <TargetFramework>net46</TargetFramework>
+    <AssemblyName>Microsoft.ApplicationInsights.EtwCollector.Tests</AssemblyName>
+    <TargetFrameworks>net46;netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
```
Comment (cijothomas) [inline LOGGING/test/EtwCollector.Tests/EtwCollector.Tests.csproj:6]: This project is only for .NET Framework so no need of testing for netcore targets.
https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/LOGGING/src/EtwCollector/EtwCollector.csproj#L3

File: LOGGING/test/EtwCollector.Tests/EtwTelemetryModuleTests.cs (hunk 1 above)
Comment (cijothomas) [reply LOGGING/test/EtwCollector.Tests/EtwTelemetryModuleTests.cs:495]: Please dont add netcore tests for this proj.

Comment (eddynaka) [reply LOGGING/test/EtwCollector.Tests/EtwTelemetryModuleTests.cs:495]: ok!

File: LOGGING/test/EtwCollector.Tests/EtwCollector.Tests.csproj (hunk 3 above)
Comment (eddynaka) [reply LOGGING/test/EtwCollector.Tests/EtwCollector.Tests.csproj:6]: ok! thanks for reviewing

File: WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj (hunk 2 above)
Comment (eddynaka) [reply WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj]: thanks for reviewing! i updated the file!

**Comment by TimothyMothra** [comment]
@eddynaka Did you get your question answered about EtwTelemetryModuleTests ?

**Comment by eddynaka** [comment]
> @eddynaka Did you get your question answered about EtwTelemetryModuleTests ?

@TimothyMothra , I didn't add netcoreapp for EtwCollector because it was made for net451. (just followed the same idea as telemetryChannel.Nuget.Tests).

**Comment by TimothyMothra** [comment]
okay, i see that now. thank you for clarifying.

File: WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    <PackageReference Include="MicroBuild.Core">
+      <Version>0.3.0</Version>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="System.Diagnostics.DiagnosticSource">
+      <Version>4.6.0</Version>
+    </PackageReference>
+    <PackageReference Include="System.Memory">
+      <Version>4.5.4</Version>
```
Comment (TimothyMothra) [inline WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj:32]: Nit: please change these PackageReferences to use the condensed format: `<PackageReference Include="System.Buffers" Version="4.5.1" />`

File: WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    <PackageReference Include="MSTest.TestFramework" Version="1.4.0" />
+    <PackageReference Include="System.Buffers" Version="4.5.1" />
+    <ProjectReference Include="..\..\..\..\BASE\src\Microsoft.ApplicationInsights\Microsoft.ApplicationInsights.csproj" />
+    <ProjectReference Include="..\HostingStartup\HostingStartup.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net45'">
+    <Reference Include="System" />
+    <Reference Include="System.Net.Http" />
+    <PackageReference Include="MicroBuild.Core">
```
Comment (TimothyMothra) [inline WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj:23]: MicroBuild is not needed for Test projects. Please remove

File: LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs [hunk 6]
```diff
@@ -294,7 +294,7 @@ public void ReportsAllProperties()
                 expected.Properties.Add("EventId", TestEventSource.ComplexEventId.ToString(InvariantCulture));
                 expected.Properties.Add("EventName", nameof(TestEventSource.ComplexEvent));
                 expected.Properties.Add("ActivityId", activityId.ToString());
-                expected.Properties.Add("Keywords", "0x8000F00000000001");
+                expected.Properties.Add("Keywords", "0x0000F00000000001");
```
Comment (TimothyMothra) [inline LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs:297]: Do you know why this change was necessary?

Comment (eddynaka) [reply LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs:297]: i changed once from 0000f to 8000f because tests were failing. after that, i reverted, so i did another commit with the original value. so 8000f to 0000f

File: LOGGING/test/EventSourceListener.Tests/TraceTelemetryComparer.cs [hunk 7]
```diff
@@ -31,6 +31,9 @@ public int Compare(object x, object y)
                 return 0;
             }
 
+            var templateHash = template.GetHashCode();
+            var actualHash = template.GetHashCode();
```
Comment (TimothyMothra) [inline LOGGING/test/EventSourceListener.Tests/TraceTelemetryComparer.cs:35]: what is this for?

Comment (eddynaka) [reply LOGGING/test/EventSourceListener.Tests/TraceTelemetryComparer.cs:35]: hum. my bad. forgot to remove. will do it soon.

Comment (eddynaka) [reply LOGGING/test/EventSourceListener.Tests/TraceTelemetryComparer.cs:35]: ok! next commit will remove those!

File: WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj (hunk 5 above)
Comment (eddynaka) [reply WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj:23]: ok. next commit will remove microbuild.core

File: WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj (hunk 4 above)
Comment (eddynaka) [reply WEB/Src/HostingStartup/HostingStartup.Tests/HostingStartup.Tests.csproj:32]: ok. next commit will condensate!

File: LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs (hunk 6 above)
Comment (TimothyMothra) [reply LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs:297]: This caught my attention because all we changed here was the framework of the test project, but this suggests that there was a change in how the product was setting values on the telemetry.
I don't have the context on what this keyword is used for.
@cijothomas did you notice this change?

Comment (cijothomas) [reply LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs:297]: the test was previously run only for 1.1. Is it failing for 1.1 now ? or failing for 2.1 and 3.1 only?

Comment (eddynaka) [reply LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs:297]: locally its working for all, and, if i'm not mistaken, it was passing in server build as well

Comment (TimothyMothra) [reply LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs:297]: Stepping back,
I see that in the CSPROJ we changed `netcoreapp1.0` to `netcoreapp1.1;netcoreapp2.1;netcoreapp3.1`

I spoke to Dmitry and he said that for EventKeywords, only the last few characters matter. He said the first character the object Type for this value.

@eddynaka can you test this locally with netcoreapp1.0?
If you can verify that netcore 1.0 passed with "0x0" then I think we can attribute this change to the changing frameworks and it's safe to merge!

Comment (eddynaka) [reply LOGGING/test/EventSourceListener.Tests/EventSourceTelemetryModuleTests.cs:297]: with 1.0 we need the 0x0 value. so, in develop, the test is passing.
